application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadAutoArchiver.java

```diff
@@ -142,4 +121,20 @@ private static boolean shouldBeArchived(MessageChannel channel, Instant archiveA
 
         return lastActivity.isBefore(archiveAfterMoment);
     }
+
+    private void handleArchiveFlow(ThreadChannel threadChannel, MessageEmbed embed) {
+        Consumer<Throwable> handleFailure = error -> {
+            if (error instanceof ErrorResponseException) {
+                logger.warn("Unknown error occurred during help thread auto archive routine",
+                        error);
+            }
+        };
+
+        threadChannel.getGuild()
+            .retrieveMemberById(threadChannel.getOwnerIdLong())
+            .flatMap(author -> threadChannel.sendMessage(author.getAsMention()).addEmbeds(embed))
+            .flatMap(any -> threadChannel.getManager().setArchived(true))
+            .queue(any -> {
```

Comment:
isnt there also an `onError(...)` or sth, so that u dont have to do it like that?

Reply:
Do you mean  `onErrorMap` or  `onErrorFlatMap` ? I had some trouble working that out. The function expects same return type as the top `flatMap` or `map`. Could you perhaps provide an example perhaps?

From Docs:
![image](https://github.com/Together-Java/TJ-Bot/assets/61616007/6a3cca1f-8f42-4ac2-8a73-f816c5f0963c)

Reply:
ah, its for mapping. well, then not. urs is fine then